packages/react/src/hooks/useResolvedKPITokens.ts

```diff
@@ -0,0 +1,77 @@
+import { Fetcher, ResolvedKPIToken } from "@carrot-kpi/sdk";
+import { type Address, usePublicClient } from "wagmi";
+import { useEffect, useState } from "react";
+import { useIPFSGatewayURL } from "./useIPFSGatewayURL";
+import { usePreferDecentralization } from "./usePreferDecentralization";
+
+interface ResolvedKPITokensParams {
+    blacklisted?: Address[];
+}
+
+export function useResolvedKPITokens(params?: ResolvedKPITokensParams): {
+    loading: boolean;
+    resolvedKPITokens: ResolvedKPIToken[];
+} {
+    const publicClient = usePublicClient();
+    const preferDecentralization = usePreferDecentralization();
+    const ipfsGatewayURL = useIPFSGatewayURL();
+    const [resolvedKPITokens, setResolvedKPITokens] = useState<
+        ResolvedKPIToken[]
+    >([]);
+
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        let cancelled = false;
+        async function fetchData(): Promise<void> {
+            if (!cancelled) setLoading(true);
+
+            try {
+                const kpiTokens = await Fetcher.fetchKPITokens({
+                    publicClient,
+                    preferDecentralization,
+                    blacklisted: params?.blacklisted,
+                });
+
+                await Promise.allSettled(
+                    Object.values(kpiTokens).map(async (kpiToken) => {
+                        try {
+                            const resolved = (
+                                await Fetcher.resolveKPITokens({
+                                    ipfsGatewayURL,
+                                    kpiTokens: [kpiToken],
+                                })
+                            )[kpiToken.address];
+                            if (!resolved) return;
+                            if (!cancelled)
+                                setResolvedKPITokens((previousState) => [
```

Comment:
I'm wondering if any time based sorting to display KPI tokens in the campaigns list is perfirmed on the campaigns page or not, because this can result in potentially out of order resolutions.